system/core/pages.php

```diff
@@ -95,8 +95,10 @@ function newPage() {
 		html = '<table class="section"><tr><th colspan="2"><?=__('create-new-page').'<br/>'.__('instruction-new-page')?></td><tr>';
 		// TODO [LRM]: find better way to set default new page type.
 		html+= '<tr><th><?=__('type')?></th><td><select id="newPageType" name="newPageType">' + '<?php foreach($types as $type => $datatypeName) echo '<option value="'.$type.'"'.($type=='textpage' ? 'selected="selected"' : '').'>'.$datatypeName.'</option>'; ?>' + '</select></td></tr>';
-		html+= '<tr><th><?=__('pagename')?></th><td><input type="text" id="newPagename" value="<?=$pagename?>" onblur="validatePagename(this);" onkeyup="validatePagename(this); document.getElementById(\'newPageSubmit\').disabled = this.value ? false : true;"/></td></tr>';
-		html+= '<tr><td></td><td><input type="checkbox" id="newPagePrivate" name="newPagePrivate"/> <?=__('private-page')?></td></tr>';
+		var infoPageName = '<?=str_replace('\'', '\\\'', makeInfoButton('help-page-name'));?>';
```

Comment:
This str_replace here probably does not catch all cases (what about backslashes?). It seems that the recommended and proper way to escape a string for javascript from PHP is to use `json_encode()` on it (which I think also generates the surrounding quotes). See https://stackoverflow.com/a/169035/740048